crossbar/router/unisocket.py

```diff
@@ -0,0 +1,150 @@
+#####################################################################################
+#
+#  Copyright (C) Tavendo GmbH
+#
+#  Unless a separate license agreement exists between you and Tavendo GmbH (e.g. you
+#  have purchased a commercial license), the license terms below apply.
+#
+#  Should you enter into a separate license agreement after having received a copy of
+#  this software, then the terms of such license agreement replace the terms below at
+#  the time at which such license agreement becomes effective.
+#
+#  In case a separate license agreement ends, and such agreement ends without being
+#  replaced by another separate license agreement, the license terms below apply
+#  from the time at which said agreement ends.
+#
+#  LICENSE TERMS
+#
+#  This program is free software: you can redistribute it and/or modify it under the
+#  terms of the GNU Affero General Public License, version 3, as published by the
+#  Free Software Foundation. This program is distributed in the hope that it will be
+#  useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+#  See the GNU Affero General Public License Version 3 for more details.
+#
+#  You should have received a copy of the GNU Affero General Public license along
+#  with this program. If not, see <http://www.gnu.org/licenses/agpl-3.0.en.html>.
+#
+#####################################################################################
+
+from __future__ import absolute_import
+
+import txaio
+txaio.use_twisted()
+
+from twisted.internet.protocol import Factory, Protocol
+
+__all__ = (
+    'UniSocketServerProtocol',
+    'UniSocketServerFactory',
+)
+
+
+class UniSocketServerProtocol(Protocol):
+    """
+    """
+
+    log = txaio.make_logger()
+
+    def __init__(self, factory, addr):
+        self._factory = factory
+        self._addr = addr
+        self._proto = None
+        self._data = b''
+
+    def dataReceived(self, data):
+
+        if self._proto:
+            # we already determined the actual protocol to speak. just forward received data
+            self._proto.dataReceived(data)
+        else:
+            if data[0] == b'\x7F':
```

Comment:
This will need to be `data[0:1]` for Python 3 (since indexing bytes on Py3 returns an integer, not a byte, but slicing returns bytes)